upp/stages/resampling.py

```diff
@@ -86,6 +95,23 @@ def pdf_select_func(self, jets, component):
         idx = random.choices(np.arange(len(jets)), weights=probs, k=num_samples)
         return idx
 
+    def pdf_upscaled_select_func(self, jets, component):
```

Comment:
It might be cleaner to just re-use the `pdf_select_func` here, we just need a couple of 

```python
bins = self.config.flag_bins
if self.upscale_pdf:
    bins = [subdivide_bins(b, self.upscale_pdf) for b in bins]
```
and then similar logic for the source and target pdfs. This will help prevent code duplication, especially as we plan to add upscaling for `countup` also.

Reply:
I do so in the new rewrite